src/encoding_object.cpp

```diff
@@ -24,7 +24,7 @@ Value EncodingObject::encode(Env *env, EncodingObject *orig_encoding, StringObje
         if (unicode_codepoint < 0) {
             StringObject *message;
 
-            if (num() != Encoding::UTF_8) {
+            if (name()->string() != "UTF-8") {
```

Comment:
this should be reverted, this was done in an attempt to reduce/eliminate using `num()` (enum) but that effort was abandoned